webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/checks/migration/MigrateChangeValidationEnforceSchema.scala

```diff
@@ -0,0 +1,128 @@
+ * the following Additional permissions:
+ * Notwithstanding to the terms of section 5 (5. Conveying Modified Source
+ * Versions) and 6 (6. Conveying Non-Source Forms.) of the GNU General
+ * Public License version 3, when you create a Related Module, this
+ * Related Module is not considered as a part of the work and may be
+ * distributed under the license agreement of your choice.
+ * A "Related Module" means a set of sources files including their
+ * documentation that, without modification of the Source Code, enables
+ * supplementary functions or services in addition to those offered by
+ * the Software.
+ *
+ * Rudder is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Rudder.  If not, see <http://www.gnu.org/licenses/>.
+
+ *
+ *************************************************************************************
+ */
+
+package bootstrap.liftweb.checks.migration
+
+import bootstrap.liftweb.BootstrapChecks
+import bootstrap.liftweb.BootstrapLogger
+import com.normation.errors.RudderError
+import com.normation.rudder.db.Doobie
+import com.normation.zio.UnsafeRun
+import doobie.ConnectionIO
+import doobie.Transactor
+import doobie.implicits._
+import doobie.util.fragment.Fragment
+import zio._
+import zio.interop.catz._
+
+/*
+ * Ths migration applies a change in the schema of the change requests and workflow table :
+ *  - add not null constraint to id, state columns in the workflow table
+ *  - add not null constraint to content column in the change requests table
+ */
+class MigrateChangeValidationEnforceSchema(
+    doobie: Doobie
+) extends BootstrapChecks {
+
+  import doobie._
+
+  val msg: String = "change requests and workflow columns that should be not null (id, state, content)"
+
+  private def changeRequestTableName: String   = "changerequest"
+  protected def changeRequestTable:   Fragment = Fragment.const(changeRequestTableName)
+  private def workflowTableName:      String   = "workflow"
+  protected def workflowTable:        Fragment = Fragment.const(workflowTableName)
+
+  override def description: String =
+    "Check if changerequest(content), workflow(id, state) have a not null constraint, otherwise migrate these columns"
+
+  private def alterTableStatement(table: Fragment, column: Fragment): Fragment = {
+    sql"alter table $table alter column $column set not null"
+  }
+
+  private def shouldMigrateColumn(tableName: String, columnName: String): ConnectionIO[Boolean] = {
+    sql"""
+      select count(*)
+      from information_schema.columns
+      where table_name = $tableName
+      and column_name = $columnName
+      and is_nullable = 'YES'
+    """.query[Int].unique.map(_ > 0)
```

Comment:
This method has been written several times.
Shouldn't it be time to put it in a `DbCommonMigration` or something?